cmake/toolchain/zephyr/0.11/target.cmake

```diff
@@ -1,36 +1,32 @@
 # SPDX-License-Identifier: Apache-2.0
 
-# Determine if we have an IAMCU toolchain or not.
-if(CONFIG_X86_IAMCU)
-  set(CROSS_COMPILE_TARGET_x86  i586-zephyr-elfiamcu)
-else()
-  set(CROSS_COMPILE_TARGET_x86 i586-zephyr-elf)
-endif()
-
 set(CROSS_COMPILE_TARGET_arm         arm-zephyr-eabi)
 set(CROSS_COMPILE_TARGET_nios2     nios2-zephyr-elf)
 set(CROSS_COMPILE_TARGET_riscv   riscv64-zephyr-elf)
 set(CROSS_COMPILE_TARGET_mips     mipsel-zephyr-elf)
 set(CROSS_COMPILE_TARGET_xtensa   xtensa-zephyr-elf)
 set(CROSS_COMPILE_TARGET_arc         arc-zephyr-elf)
 set(CROSS_COMPILE_TARGET_x86_64   x86_64-zephyr-elf)
+set(CROSS_COMPILE_TARGET_x86        i586-zephyr-elf)
 
 set(CROSS_COMPILE_TARGET ${CROSS_COMPILE_TARGET_${ARCH}})
 set(SYSROOT_TARGET       ${CROSS_COMPILE_TARGET})
 
 set(CROSS_COMPILE ${TOOLCHAIN_HOME}/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+set(SYSROOT_DIR   ${TOOLCHAIN_HOME}/${SYSROOT_TARGET}/${SYSROOT_TARGET})
 
 if("${ARCH}" STREQUAL "xtensa")
-  set(SYSROOT_DIR ${TOOLCHAIN_HOME}/${SYSROOT_TARGET})
+  set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/${SOC_NAME}/${SYSROOT_TARGET})
   set(TOOLCHAIN_INCLUDES
     ${SYSROOT_DIR}/include/arch/include
     ${SYSROOT_DIR}/include
     )
 
   LIST(APPEND TOOLCHAIN_LIBS hal)
   LIST(APPEND LIB_INCLUDE_DIR -L${SYSROOT_DIR}/lib)
+
+  set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/${SOC_NAME}/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
 elseif("${ARCH}" STREQUAL "x86_64")
   list(APPEND TOOLCHAIN_C_FLAGS -mx32)
   list(APPEND TOOLCHAIN_LD_FLAGS -mx32)
 endif()
```

Comment:
Please treat these variables as immutable, e.g.

```
if(ARCH STREQUAL xtensa)
set(CROSS_COMPILE bla bla)
else()
set(CROSS_COMPILE yada yada)
endif()

```

It is easier to debug and read when the variables are not variable.

Reply:
I have updated the file. Is this what you want?